src/linear_algebra/finsupp.lean

```diff
@@ -822,6 +822,46 @@ end prod
 
 end finsupp
 
+section fintype
+
+variables (α M R : Type*) [semiring R] [add_comm_monoid M] [module R M] (v : α → M) [fintype α]
+
+/-- A variant of `finsupp.total` that is defined on fintype indexed vectors. -/
+protected def fintype.total : (α → R) →ₗ[R] M :=
+{ to_fun := λ f, ∑ i, f i • v i,
+  map_add' := λ f g, by { simp_rw [← finset.sum_add_distrib, ← add_smul], refl },
+  map_smul' := λ r f, by { simp_rw [finset.smul_sum, smul_smul], refl } }
```

Comment:
I wonder if we can build this out of:
```
/-- `finset.sum as a linear map -/
protected def finset.lsum (s : finset α) : (α → M) →ₗ[R] M :=
{ to_fun := λ f, ∑ i in s, f i,
  map_add' := λ f g, by { simp_rw [← finset.sum_add_distrib], refl },
  map_smul' := λ r f, by { simp_rw [finset.smul_sum], refl } }
```

Reply:
If we have the instances 
```
instance : is_scalar_tower R (α → R) (α → M) := sorry
instance : is_scalar_tower R (α → R) (α → R) := sorry
```
We should be able to build it from
```
(finset.univ.lsum α R M).comp ((linear_map.to_span_singleton (α → R) (α → M) v).restrict_scalars R)
```
But I'm not sure if it is better?

Reply:
Oh, we already have `linear_map.lsum`